pkg/functions/functions.go

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package functions
+
+import (
+	"path/filepath"
+	"github.com/projectriff/riff-cli/pkg/osutils"
+)
+
+func FunctionNameFromPath(path string) string {
+	if osutils.IsDirectory(path) {
+		return filepath.Base(path)
```

Comment:
what happens if path is `.`

Reply:
using `filepath.Abs(path)` should get an absolute path even from a `.`

Reply:
Agree